src/chrome_trace/chrome_trace.mli

```diff
@@ -18,6 +18,54 @@ module Json : sig
     | Object of (string * t) list
 end
 
+module Event : sig
+  type t
+
+  module Timestamp : sig
+    type t
+
+    val of_float_seconds : float -> t
+
+    val now : unit -> t
+  end
+
+  module Id : sig
+    type t =
+      | Int of int
+      | String of string
+  end
+
+  type common
+
+  val common :
+       ?tts:Timestamp.t
+    -> ?cname:string
+    -> ?cat:string list
+    -> ts:Timestamp.t
+    -> name:string
+    -> pid:int
```

Comment:
> Presumably you're intending this library to be a fully general library for Chrome_trace generation

Yes, I'd like the library to just reflect the spec (even the hairy bits).

> Maybe there should be a wrapper on top that "simplifies" the API, filling in these "useless" args with the values we always pass.

We could also just use real values for these fields. There's `Thread.id` and `Unix.getpid` after all. It won't do much good now, but we could imagine allowing DAP actions to add their own trace events. In which case, distinguishing different processes would be handy. Or I could bring back a simplified `Stats` module that wraps parts of the api.

Reply:
I'll be surprised if passing real values can be useful, but I guess if it's constant across a run of dune it can't hurt too much. In general, being faithful to the Chrome API seems like a questionable design goal to me because their API seems very much tailored to their needs, so to make something look the way you want on the report you occasionally have to "lie" in the trace.

Reply:
Actually, how about I'll just make them optional and default to 0? This way it's convenient for us and if someone wants to customize these, it remains an option.